drivers/bm_manchester/bm_manchester.c

```diff
@@ -152,7 +152,7 @@ bm_manchester_ret_t bm_manchester_align(uint8_t *rx_byte, volatile uint8_t *buf,
 		*buf_idx = 0;
 		delimiter_ctr = 0;
 		payload_ctr = 0;
-		return ret;
+		goto out;
```

Comment:
I added more details in the goto below.
For cases like this, i totally appreciate avoiding the early return. The way I do this, which is similar in function to a goto, but with fewer chances of error, is to wrap the block of code in a:
```
do {

} while (0);
```

And then use `break` when you want to get out of the loop. In this case, you can pretty much wrap most of the function in the do while 0 and break instead of goto

Reply:
Deprecated :) had to remove the manchester specific file. We ending up stripping a lot of complexity and logic of the encoding and parsing.